ue5osc/__init__.py

```diff
@@ -1,21 +1,108 @@
+from pathlib import Path
+from time import sleep
+
+from PIL import Image
 from pythonosc import udp_client
+from pythonosc.osc_server import BlockingOSCUDPServer
+import threading
+
+from ue5osc.osc_dispatcher import OSCMessageReceiver
+
+
+class Communicator:
+    """This handles interaction between the UE5 environment and the a program."""
+
+    def __init__(self, ip: str, client_port: int, server_port: int, directory: str):
+        """Initialize OSC client and server."""
+        self.path = Path(directory)
+        self.img_number = 0
+
+        self.message_handler = OSCMessageReceiver()
+        self.server = BlockingOSCUDPServer(
+            (ip, server_port), self.message_handler.dispatcher
+        )
+        self.client = udp_client.SimpleUDPClient(ip, client_port)
+
+        self.server_thread = threading.Thread(target=self.server.serve_forever)
+        print(f"Server is Listening to {ip}:{server_port} ...")
+        self.server_thread.start()
+
+    def close_osc(self):
+        """Closes the OSC server and joins the server."""
+        print("Closing the server...")
+        sleep(1)
+        self.server.shutdown()
+        self.server_thread.join()
+        print("Server Closed")

```

Comment:
Remove.

Reply:
Should I completely delete the close_osc function?

Reply:
No, just the sleep and print. I must of misclicked.